### PR TITLE
gettext fixes

### PR DESCRIPTION
**Pull request by brooksdavis**

### CODE REVIEW

File: Mk/bsd.port.mk [hunk 1]
```diff
@@ -3247,6 +3247,20 @@ do-patch:
 			${SH} ${SCRIPTSDIR}/do-patch.sh
 .    endif
 
+.    if !target(run-cheri-gnulib-fixup)
+run-cheri-gnulib-fixup:
+# Work around gnulib insisting on providing it's own stdint.h which (among
+# other things) redefines (u)intptr_t to (unsigned) long with a hard to
+# mix of typedefs and #defines.  As this is mostly pointless, replace with
```
[the remark this answers is not in the document]
Comment (brooksdavis) [reply Mk/bsd.port.mk]: hopefully it makes more sense with the missing word "debug"?

[the remark this answers is not in the document]
Comment (jrtc27) [reply Mk/bsd.port.mk]: Yep, thanks

**Review by jrtc27 — APPROVED** [review]
Changes look fine to me now. Duplicating the patch is a little sad but I know how clunky it is to reuse patches across ports, so for something this simple so be it.

**Comment by brooksdavis** [comment]
> Changes look fine to me now. Duplicating the patch is a little sad but I know how clunky it is to reuse patches across ports, so for something this simple so be it.

**Comment by brooksdavis** [comment]
arg

**Comment by brooksdavis** [comment]
> Changes look fine to me now. Duplicating the patch is a little sad but I know how clunky it is to reuse patches across ports, so for something this simple so be it.

Unconditionally using ../gettext-runtime in the patch paths allowed me to use EXTRA_PATCHES in gettext-tools so

File: devel/gettext-tools/Makefile [hunk 2]
```diff
@@ -11,9 +11,6 @@ WWW=		https://www.gnu.org/software/gettext/
 LICENSE=	GPLv3+
 LICENSE_FILE=	${WRKSRC}/../COPYING
 
-BROKEN_purecap=	msgfmt is broken for CheriABI
```
Comment (kwitaszczyk) [inline devel/gettext-tools/Makefile:14]: If we want to use CheriABI msgfmt, we should revert 5af5781a4615a5a3961ef7c1aa5d010ec94fbef9.

**Pull request merged**